feat: add `lapack/base/disnan` (#12289)
stdlib-js/stdlib: lib/node_modules/@stdlib/lapack/base/disnan/README.md

<!--

@license Apache-2.0

Copyright (c) 2026 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# disnan

> > LAPACK auxiliary routine to test a double-precision floating-point numeric value for `NaN`.

Reviewer: ```suggestion
> LAPACK auxiliary routine to test whether a double-precision floating-point number is `NaN`.
```

<section class="usage">

## Usage

```javascript
var disnan = require( '@stdlib/lapack/base/disnan' );
```

> #### disnan( din )

Reviewer: All the same comments.

Tests a double-precision floating-point numeric value for `NaN`.

```javascript
var bool = disnan( NaN );
// returns true

bool = disnan( 5.0 );
// returns false
```

The function has the following parameters:

-   **din**: double-precision floating-point number to test.

</section>

<!-- /.usage -->

<section class="notes">

## Notes

-   `disnan()` corresponds to the [LAPACK][lapack] auxiliary routine [`disnan`][lapack-disnan].

> -   `disnan` tests for `NaN` by internally calling [`dlaisnan`][lapack-dlaisnan] with the same argument passed twice (i.e., `dlaisnan( din, din )`). This approach avoids issues with compiler optimization that might incorrectly evaluate `NaN !== NaN` as `false`.

Reviewer: ```suggestion
```

This info isn't super relevant for the average user.
